core/src/main/java/org/infinispan/configuration/cache/TransactionConfigurationBuilder.java

```
@@ -327,6 +332,18 @@ public void validate() {
       if (attributes.attribute(TRANSACTION_MODE).get() == TransactionMode.TRANSACTIONAL && !cacheMode.isSynchronous()) {
          throw log.unsupportedAsyncCacheMode(cacheMode);
       }
+      if (attributes.attribute(TRANSACTION_MODE).get() == TransactionMode.TRANSACTIONAL &&
+            attributes.attribute(LOCKING_MODE).get() == LockingMode.OPTIMISTIC &&
+            !attributes.attribute(USE_EAGER_LOCKING).get() && //this is pessimistic transaction
+            getBuilder().locking().isolationLevel() == IsolationLevel.REPEATABLE_READ &&
+            cacheMode != CacheMode.INVALIDATION_SYNC && //write-skew only for REPL, DIST and LOCAL (async is not supported in tx mode)
+            !getBuilder().invocationBatching().isEnabled()) { //batching can live without WSC
+         getBuilder().locking().writeSkewCheck(true);
+         getBuilder().versioning().enable().scheme(VersioningScheme.SIMPLE);
```

@rvansa @danberindei 
I prefer to throw an exception, but that will mess up everything. Like, in 9.0.CR3 all your config is fine and in 9.0.Final, you have exceptions everywhere (cc @tristantarrant). what do you wanna do?